homeassistant/components/alert.py

```diff
@@ -0,0 +1,294 @@
+                       alert[CONF_NOTIFIERS],
+                       alert.get(CONF_CAN_ACK, DEFAULT_CAN_ACK),
+                       alert.get(CONF_BACKOFF, DEFAULT_BACKOFF))
+        all_alerts[entity.entity_id] = entity
+
+    descriptions = yield from hass.loop.run_in_executor(
+        None, load_yaml_config_file, os.path.join(
+            os.path.dirname(__file__), 'services.yaml'))
+    descriptions = descriptions.get(DOMAIN, {})
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_TURN_OFF, async_handle_alert_service,
+        descriptions.get(SERVICE_TURN_OFF), schema=ALERT_SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_TURN_ON, async_handle_alert_service,
+        descriptions.get(SERVICE_TURN_ON), schema=ALERT_SERVICE_SCHEMA)
+    hass.services.async_register(
+        DOMAIN, SERVICE_TOGGLE, async_handle_alert_service,
+        descriptions.get(SERVICE_TOGGLE), schema=ALERT_SERVICE_SCHEMA)
+
+    for alert in all_alerts.values():
+        yield from alert.async_update_ha_state()
+
+    return True
+
+
+class Alert(ToggleEntity):
+    """Representation of an alert."""
+
+    def __init__(self, hass, name, entity, state, repeat, skip_first,
+                 notifiers, can_ack, backoff):
+        """Initialize the alert."""
+        self.hass = hass
+        self._name = name
+        self._alert_state = state
+        self._delay = timedelta(minutes=repeat)
+        self._skip_first = skip_first
+        self._notifiers = notifiers
+        self._can_ack = can_ack
+        self._backoff = backoff
+
+        self._next_delay = None
+        self._firing = False
+        self._ack = False
+        self._cancel = None
+        self.entity_id = ENTITY_ID_FORMAT.format(slugify(name))
+
+        event.async_track_state_change(hass, entity,
+                                       self.watched_entity_change)
+
+    @property
+    def name(self):
+        """Return the name of the alert."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the alert status."""
+        if self._firing:
+            if self._ack:
+                return STATE_OFF
+            return STATE_ON
+        return STATE_IDLE
+
+    @property
+    def hidden(self):
+        """Hide the alert when it is not firing."""
+        return not self._can_ack or not self._firing
+
+    @asyncio.coroutine
+    def watched_entity_change(self, entity, from_state, to_state):
+        """Determine if the alert should start or stop."""
+        _LOGGER.debug('Watched entity (%s) has changed.', entity)
+        if to_state.state == self._alert_state and not self._firing:
+            return self.begin_alerting()
+        if to_state.state != self._alert_state and self._firing:
+            return self.end_alerting()
+
+    @asyncio.coroutine
+    def begin_alerting(self):
+        """Begin the alert procedures."""
+        _LOGGER.info('Begining Alert: %s', self._name)
```

Comment:
Also, this can be `debug` level because the state change will already be logged as 'info' to indicate something happened.